src/location.rs

```diff
@@ -1,10 +1,43 @@
 use bevy::prelude::*;
+use crate::direction::{Direction};
 
-pub type XCoord = u8;
-pub type YCoord = u8;
+type Coord = u8;
 
-#[derive(Component, Default, Debug)]
+const MAX_X: Coord = 15;
+const MAX_Y: Coord = 8;
+
+#[derive(Component, Debug)]
 pub struct Location {
-    x: XCoord,
-    y: YCoord,
+    x: Coord,
+    y: Coord,
+}
+
+impl Location {
+    pub fn new() -> Self {
+        Self { x: MAX_X / 2, y: MAX_Y / 2 }
+    }
+
+    pub fn move_by_one(&mut self, dir: &Direction) -> Result<(), ()> {
+        match dir {
+            Direction::Up => try_add(&mut self.y, -1, MAX_Y),
+            Direction::Down => try_add(&mut self.y, 1, MAX_Y),
+            Direction::Left => try_add(&mut self.x, -1, MAX_X),
+            Direction::Right => try_add(&mut self.x, 1, MAX_X),
+        }
+    }
+
+    pub fn get_translation(&self) -> Vec3 {
```

Comment:
I think this is a concern of the Board? We might need a board module. Perhaps there are some parameters like shape and size that mean the board size and wrap-around behaviour is not constant.